### PR TITLE
Add test code coverage

### LINKED ISSUE #83: Profile test coverage
We should profile the tests to figure out what kind of coverage we have, and identify potential targets for increased test coverage.

The outcome of this should be new issues for uncovered parts of the code base.

### PR DESCRIPTION
**Pull request by DanSava**
**Issue**

Closes #83

### CODE REVIEW

**Review by oyvindeide — APPROVED** [review]
LGTM!

File: .github/workflows/coverage.yml [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+     - 'version-**'
+   tags: "*"
+ pull_request:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: ${{ github.ref != 'refs/heads/main' }}
+
+env:
+  ERT_SHOW_BACKTRACE: 1
```
Comment (oyvindeide) [inline .github/workflows/coverage.yml:16]: This is not needed and has no effect

File: .github/workflows/coverage.yml [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+  NO_PROJECT_RES: 1
```
Comment (oyvindeide) [inline .github/workflows/coverage.yml:17]: Dont think this is needed?

Comment (DanSava) [reply .github/workflows/coverage.yml:17]: The `tests/test_egg_simulation.py::test_run_egg_model` will run and fail without this env var.

**Pull request merged**